crates/sui-types/src/gas_model/gas_v2.rs

```diff
@@ -418,12 +418,19 @@ pub(crate) fn check_gas_balance(
     gas_budget: u64,
     cost_table: &SuiCostTable,
 ) -> UserInputResult {
-    // 1. Gas object has an address owner.
+    // 1. All gas objects have an address owner
     if !(matches!(gas_object.owner, Owner::AddressOwner(_))) {
         return Err(UserInputError::GasObjectNotOwnedObject {
             owner: gas_object.owner,
         });
     }
+    for gas_object in &more_gas_objs {
```

Comment:
This should be version-gated, but it's also the case that any tx that hits this is doing something we don't want. Not sure how to handle this.

Reply:
if there's no "invalid' txes in the past then we are good ...

Reply:
if not considering versioning, I'd do sth like 
```
for gas_object in &more_gas_objs.chain(iter::once(gas_object)) {
...
}
```
gas_object chains more_gas_objs